services/src/main/java/org/keycloak/protocol/saml/SamlService.java

```diff
@@ -607,6 +619,14 @@ private static void addKeyInfo(StringBuilder target, RsaKeyMetadata key, String
           key.getKid(), PemUtils.encodeCertificate(key.getCertificate()), purpose, false));
     }
 
+    private boolean isClientProtocolCorrect(ClientModel clientModel) {
+        if (clientModel.getProtocol().equals(SamlProtocol.LOGIN_PROTOCOL)) {
```

Comment:
Minor comment, but is is it possible to change this to rather change order to:
````
if (SamlProtocol.LOGIN_PROTOCOL.equals(clientModel.getProtocol()))
````
I have some worries about possible NullPointerException (as there is some chance that ClientModel.getProtocol() may return null.